### PR TITLE
Introduce native OS linkage example and README

### PR DESCRIPTION
**Pull request by dkelosky**
This introduces a `/wto` endpoint to demonstrate a JNI example within the sample spring boot API service.  The example uses OS linkage to show how to call a traditional z/OS interface, WTO.

### CODE REVIEW

**Comment by vvvlc** [comment]
I understand that purpose is to demonstrate how to do  OS LINKAGE and `wto` was selected as an example.
[`__console2`](https://www.ibm.com/support/knowledgecenter/SSLTBW_2.3.0/com.ibm.zos.v2r3.bpxbd00/consol2.htm) from standard library can issue WTO as well, in order to leverage `__console2` JNI code must translate utf8 to ebcdic.

**Comment by dkelosky** [comment]
yes, WTO was an arbitrary example - do you think some other service would be better understood ?

File: docs/zos-native-os-linkage.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# z/OS Native OS Linkage Example
+
+These steps describe how to build and test this sample REST API in a way that
+uses JNI to call a native z/OS service - WTO.  You can follow these steps and modify them to
+call your own assembler / C API.
+
+These instructions assume that you follow the [z/OS deployment setup](./zos-deployment).
```
Comment (tucker01) [inline docs/zos-native-os-linkage.md]: Link is broken... looks like you're missing the extension (.md)

File: docs/zos-native-os-linkage.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+## JNI Overview
+
+There are many tutorials online to describe details and examples of JNI.  
+
+At a high level, the process involves making use of the `native` keyword in a class.  After this, you run the `javah` command against the class to emit a C/C++ header file.  This emitted header file will contain function prototypes for the native method for which you must provide the implementation.
+
+Here is the example `javah` command for this project:
```
Comment (tucker01) [inline docs/zos-native-os-linkage.md]: Just an FYI, At least with oracle, etc. `javah` is no longer available I believe after JDK 8. I think it's replaced by a flag on the compiler.

File: zossrc/chdr/wtoexec.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+/**                                                                                         
+ * C/C++ interlanguage - "OS" meaning to call non-Language Environment conforming assembler.
+ */
+#if defined(__cplusplus) && (defined(__IBMCPP__) || defined(__IBMC__))
```
Comment (tucker01) [inline zossrc/chdr/wtoexec.h]: It's fine to keep this stuff in here, but some of this preprocessor if logic was to allow compilation off z/OS. I'm just not sure if it confuses the example?

File: zossrc/makefile [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                                                                       
+ASM = as                                                               
+ASM_FLAGS = -mrent                                                     
+                                                                       
+libwtojni.so: wtojni.o wtoexec.o                                       
+	$(CXX) $(DLL_BND_FLAGS) -o $@ $¬ > $*.bind.lst                          
+	extattr +p $@                                                         
+                                                                       
+wtojni.o: wtojni.cpp                                                   
+	$(CXX) $(DLL_CPP_FLAGS) -qlist=$*.cpp.lst -o $@ $¬
```
Comment (tucker01) [inline zossrc/makefile]: Does the `¬` work properly if you upload/download via ftp/zowe? I know we have a problem using ISPF and the `^` character.

**Comment by tucker01** [comment]
I added some comments. I think its great you guys are including this in the sample.

File: docs/zos-native-os-linkage.md (hunk 1 above)
Comment (dkelosky) [reply docs/zos-native-os-linkage.md]: Totally blaming VS Code for this one 😄 

Fixed.

File: docs/zos-native-os-linkage.md (hunk 2 above)
Comment (dkelosky) [reply docs/zos-native-os-linkage.md]: Good point - I didn't know this.  I made a note in the markdown - maybe in a future PR the `makefile` and sample command will be updated to use the flag.

File: zossrc/chdr/wtoexec.h (hunk 3 above)
Comment (dkelosky) [reply zossrc/chdr/wtoexec.h]: I agree that it confuses the example.  It seems that you need some level of #ifdef for the metal C compilation of the header and for the C++ header.  If I'm leaving some, perhaps it's best to leave this as-is to get people setup for on and off platform compilation.  What do you think?

Comment (tucker01) [reply zossrc/chdr/wtoexec.h]: Sounds good to me.

File: zossrc/makefile (hunk 4 above)
Comment (dkelosky) [reply zossrc/makefile]: yeah - I messed that up - fixed in the latest push

**Comment by dkelosky** [comment]
thanks @gejohnston for taking the time to review - I'll take a stab at addressing both of your suggestions in this PR so that your suggestions are not lost.  In time, future PRs could do more to comment on individual classes, C source, and README than my incoming updates.

**Pull request merged**